priorityq/priorityq.go

```diff
@@ -0,0 +1,87 @@
+package priorityq
```

Comment:
About naming, I think the package name could be `priority`, the `PrioritryQ` name could be `Queue`, and the Priority level could stay `Priority` or renamed to `Level`.
that way we will write `priority.Queue, priority.Level` instead of writing priority 2 times every time. 
let me know what you think :)

Reply:
The level is suited for logs. Priority is more clear.
It is true that we duplicate a bit and I know that it is not the GO-way but naming the package priority is wrong since currently, it is only a package of priority q. If it changes in the future we can rename it.
I don't mined renaming PriorityQ to Queue.
Then we have priorityq.Priority and priorityq.Queue and also rename NewPriorityQ to New.